library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -173,6 +201,60 @@ class Client() {
         }
     }
 
+    private suspend fun ffiXmtpClient(
+        options: ClientOptions?,
+        account: SigningKey,
+        appContext: Context?,
+        privateKeyBundleV1: PrivateKeyBundleV1,
+    ): FfiXmtpClient? {
+        val libXMTPClient: FfiXmtpClient? =
+            if (options != null && options.enableLibXmtpV3 && options.appContext != null) {
+                val alias = "xmtp-${options.api.env}-${account.getAddress()}"
+
+                val dbDir = File(appContext?.filesDir?.absolutePath, "xmtp_db")
+                dbDir.mkdir()
+                val dbPath: String = dbDir.absolutePath + "/$alias.db3"
+
+                val sharedPreferences =
+                    options.appContext.getSharedPreferences("XMTPPreferences", Context.MODE_PRIVATE)
+                val dbEncryptionKey = sharedPreferences.getString(alias, null)
+
+                val retrievedKey = if (dbEncryptionKey != null) {
+                    Base64.decode(dbEncryptionKey)
+                } else {
+                    val editor = sharedPreferences.edit()
+                    val key: ByteArray = SecureRandom().generateSeed(32)
+                    editor.putString(alias, Base64.encode(key))
+                    editor.apply()
+                    key
+                }
+
+                createClient(
+                    logger = logger,
+                    host = "http://10.0.2.2:5556",
+                    isSecure = false,
+                    db = dbPath,
+                    encryptionKey = retrievedKey,
+                    accountAddress = account.getAddress(),
+                    // Figure out where the keys came from
+                    //    NONE,
+                    //    STATIC, // someone passed in a bundle
+                    //    NETWORK, // v2 keys already on the network
+                    //    KEY_GENERATOR; // I created them
+                    legacyIdentitySource = LegacyIdentitySource.NETWORK,
+                    legacySignedPrivateKeyProto = privateKeyBundleV1.toV2().toByteArray()
+                )
+            } else {
+                null
+            }
+
+        libXMTPClient?.textToSign()?.let {
```

Comment:
I think I may have led you astray here. You'll need to call register identity one way or the other here. If there is `textToSign` you'll call it with the signature, and if there isn't you'll call it with `null`.